bot_scripts/manage-pr-checklist.js

```diff
@@ -0,0 +1,164 @@
+    const hasChange = firstCheck || currentStatus !== newStatus
+    const logStatus = isChecklistComplete ? '✅' : '⏳'
+    const shortUrl = `${context.payload.repository.full_name}#${context.payload.pull_request.number}`
+
+    const log = context.log.child({
+        event: context.event.event,
+        action: context.payload.action,
+        account: context.payload.repository.owner.id,
+        repo: context.payload.repository.id,
+        change: hasChange,
+        prchecklist: isChecklistComplete
+    })
+
+    if(!hasChange){
+        return log.info(`😐${logStatus} ${shortUrl}`) 
+    }
+
+    try {
+        await context.github.repos.createStatus(context.repo({
+          sha: context.payload.pull_request.head.sha,
+          state: newStatus,
+          target_url: 'https://github.com/status-im/status-github-bot.git',
+          description: isChecklistComplete ? 'ready for merge' : 'PR Checklist is incomplete',
+          context: 'PRChecklist'
+        }))
+    
+        log.info(`💾${logStatus} ${shortUrl}`)
+    } catch (err) {
+        log.error(err)
+    }
+}
+
+
+async function getCurrentStatus (context) {
+    const {data: {statuses}} = await context.github.repos.getCombinedStatusForRef(context.repo({
+      ref: context.payload.pull_request.head.sha
+    }))
+  
+    return (statuses.find(status => status.context === 'PRChecklist') || {}).state
+}
+
+async function createOrEditChecklist(context, checkList, header){
+    let owner = context.payload.repository.owner.login
+    let repo = context.payload.repository.name
+    let number = context.payload.pull_request.number
+    if (checkList && checkList.length > 0) {
+        let body = "<!--prchecklist--> \n" + header + "\n"
+        for (let key of checkList) {
+            body += "- [ ] " + key + "\n"
+        }
+        await context.github.issues.createComment({ owner, repo, number, body})
+    }
+}
+
+async function verifyChecklist(context, settings){
+    let isChecklistComplete = true
+    let firstCheck = false
+    let {found, body, comment_id} = await getCheckList(context)
+    if(found){
+        if(body){
+            for(let str of body){
+                let res = str.match(/(-\s\[(\s)])(.*)/gm)
+                if(res != null){
+                    isChecklistComplete = false
+                    break
+                }
+            }
+        }else{
+            isChecklistComplete = false
+        }
+    }else {
+        await createOrEditChecklist(context, settings.checklist, settings.title)
+        isChecklistComplete = false
+        firstCheck = true
+    }
+    return {isChecklistComplete, firstCheck}
+}
+
+async function getCheckList(context){
+    try{
+        let owner = context.payload.repository.owner.login
+        let repo = context.payload.repository.name
+        let number = context.payload.pull_request.number
+        let comments = await context.github.issues.getComments({ owner, repo, number});
```

Comment:
Maybe we should use `github.paginate` to make sure we don't ask for a huge number of comments in one go. There are examples available in other scripts.